### ISSUE TITLE
BotApp.java:57-59: Refactor this class: move actual logic...

### ISSUE THREAD

**Issue by 0pdd** [g4s8/ghman#2]
The puzzle `1-b3595789` from #1 has to be resolved: 

https://github.com/g4s8/ghman/blob/a8f587ea02446832a0de164a20b03d7667218015/src/main/java/com/g4s8/ghman/bot/BotApp.java#L57-L59

The puzzle was created by @g4s8 on 22-May-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
Job [#2](https://github.com/g4s8/ghman/issues/2) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) everybody who has role `DEV` is banned at [#2](https://github.com/g4s8/ghman/issues/2); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
The job [#2](https://github.com/g4s8/ghman/issues/2) assigned to @samihibrahim[/z](https://www.0crat.com/u/samihibrahim), here is [why](https://www.0crat.com/footprint/CJT4R32Q3/72f40d66-29ca-4b36-a66a-0a29ebf6f5ea); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by samihibrahim** [comment]
@0crat wait while PR #21 is merged

**Comment by 0crat** [comment]
> @0crat wait while PR #21 is merged ([here](https://github.com/g4s8/ghman/issues/2#issuecomment-500051282))

@samihibrahim The impediment for [#2](https://github.com/g4s8/ghman/issues/2) was registered successfully by @samihibrahim[/z](https://www.0crat.com/u/samihibrahim)

**Comment by samihibrahim** [comment]
@g4s8 I am not sure if I broke something or what. But why PDD is not closing this issue?

**Comment by g4s8** [comment]
@samihibrahim there're few pdd errors:
```
ERROR: Puzzle src/main/java/com/g4s8/ghman/bot/TkNotifications.java:40-40 has a very short description of just 3 words while a minimum of 10 is required
ERROR: Puzzle src/main/java/com/g4s8/ghman/web/PsUserById.java:30-30 has a very short description of just 5 words while a minimum of 10 is required
ERROR: Puzzle src/main/java/com/g4s8/ghman/web/TkGitHubAuthorization.java:39-39 has a very short description of just 5 words while a minimum of 10 is required
ERROR: Puzzle src/main/java/com/g4s8/ghman/web/FkGitHubAuthRedirection.java:35-35 has a very short description of just 5 words while a minimum of 10 is required
ERROR: 4 errors, see log above
```

As I can see with `git blame` you submitted wrong-formatted puzzles which can't be processed by pdd, you need to fix 4 puzzles to close this issue.

it's my bad, I didn't configure pdd checks for Rultor and CI to see this error before merge, feel free to submit a ticket about it and I'll fix it.

**Comment by samihibrahim** [comment]
@g4s8 well, my bad. The problem is these puzzles are from different issues (#2 & #4) and both PRs are closed now. Should I create a single new PR fixing all four puzzles? or two new PR for each issue? or create a single new Issue and fix it with its own PR?

**Comment by g4s8** [comment]
@samihibrahim just create a single PR with all puzzle fixes

**Comment by 0pdd** [comment]
The puzzle `1-b3595789` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd 5 puzzles [#34](https://github.com/g4s8/ghman/issues/34), [#35](https://github.com/g4s8/ghman/issues/35), [#36](https://github.com/g4s8/ghman/issues/36), [#37](https://github.com/g4s8/ghman/issues/37), [#38](https://github.com/g4s8/ghman/issues/38) are still not solved.

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please review this job completed by @samihibrahim[/z](https://www.0crat.com/u/samihibrahim), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#2](https://github.com/g4s8/ghman/issues/2) is not in the agenda of @samihibrahim[/z](https://www.0crat.com/u/samihibrahim), can't inspect

**Comment by 0crat** [comment]
The job [#2](https://github.com/g4s8/ghman/issues/2) is now out of scope

**Comment by ammaratef45** [comment]
@samihibrahim 
>All commit messages have links to issues

this rule was violated in your PR, please confirm you would take of that in the future.

**Comment by samihibrahim** [comment]
@ammaratef45 confirmed.

**Comment by ammaratef45** [comment]
@samihibrahim Thanks

**Comment by ammaratef45** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +30 point(s) just awarded to @samihibrahim[/z](https://www.0crat.com/u/samihibrahim)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#34](https://github.com/g4s8/ghman/issues/34), [#35](https://github.com/g4s8/ghman/issues/35), [#36](https://github.com/g4s8/ghman/issues/36), [#37](https://github.com/g4s8/ghman/issues/37) are still not solved; solved: [#38](https://github.com/g4s8/ghman/issues/38).

**Comment by 0pdd** [comment]
@0pdd 3 puzzles [#34](https://github.com/g4s8/ghman/issues/34), [#35](https://github.com/g4s8/ghman/issues/35), [#36](https://github.com/g4s8/ghman/issues/36) are still not solved; solved: [#37](https://github.com/g4s8/ghman/issues/37), [#38](https://github.com/g4s8/ghman/issues/38).

**Comment by 0pdd** [comment]
@0pdd 5 puzzles [#34](https://github.com/g4s8/ghman/issues/34), [#36](https://github.com/g4s8/ghman/issues/36), [#82](https://github.com/g4s8/ghman/issues/82), [#83](https://github.com/g4s8/ghman/issues/83), [#84](https://github.com/g4s8/ghman/issues/84) are still not solved; solved: [#35](https://github.com/g4s8/ghman/issues/35), [#37](https://github.com/g4s8/ghman/issues/37), [#38](https://github.com/g4s8/ghman/issues/38).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#34](https://github.com/g4s8/ghman/issues/34), [#82](https://github.com/g4s8/ghman/issues/82), [#83](https://github.com/g4s8/ghman/issues/83), [#84](https://github.com/g4s8/ghman/issues/84) are still not solved; solved: [#35](https://github.com/g4s8/ghman/issues/35), [#36](https://github.com/g4s8/ghman/issues/36), [#37](https://github.com/g4s8/ghman/issues/37), [#38](https://github.com/g4s8/ghman/issues/38).
